tests/sensors/test_time_delta.py

```diff
@@ -71,3 +71,17 @@ def test_timedelta_sensor(self, defer_mock, should_defer):
             defer_mock.assert_called_once()
         else:
             defer_mock.assert_not_called()
+
+    @pytest.mark.parametrize(
+        "should_defer",
+        [False, True],
+    )
+    @mock.patch("airflow.models.baseoperator.BaseOperator.defer")
+    def test_wait_sensor(self, defer_mock, should_defer):
+        delta = timedelta(seconds=30)
+        op = WaitSensor(task_id="wait_sensor_check", delta=delta, dag=self.dag, deferrable=should_defer)
+        op.execute({})
+        if should_defer:
+            defer_mock.assert_called_once()
```

Comment:
Can you use `assert_called_once_with(...)`?

Reply:
Trigger class isn't set up for this unfortunately.

```
FAILED tests/sensors/test_time_delta.py::TestTimeDeltaSensorAsync::test_wait_sensor[True] - AssertionError: expected call not found.
Expected: defer(trigger=<airflow.triggers.temporal.DateTimeTrigger moment=2024-08-01 00:00:30+00:00, end_from_trigger=True>, method_name='execute_complete')
  Actual: defer(trigger=<airflow.triggers.temporal.DateTimeTrigger moment=2024-08-01 00:00:30+00:00, end_from_trigger=True>, method_name='execute_complete')
```